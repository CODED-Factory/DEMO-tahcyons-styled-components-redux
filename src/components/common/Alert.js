import React from "react";

const Alert = ({ level = "", text }) => {
  const theme = {
    background: "",
    color: ""
  };

  switch (level) {
    case "error":
      theme.background = "bg-light-red";
      theme.color = "white";
      break;
    case "info":
      theme.background = "bg-lightest-blue";
      theme.color = "navy";
      break;
    case "success":
      theme.background = "bg-light-green";
      theme.color = "dark-green";
      break;
    default:
      theme.background = "bg-light-yellow";
      theme.color = "white";
  }

  const { background, color } = theme;

  return (
    <div
      className={`flex items-center justify-center pa3 br2 mv2 ${background} ${color}`}
    >
      <span>{text}</span>
    </div>
  );
};

export default Alert;
