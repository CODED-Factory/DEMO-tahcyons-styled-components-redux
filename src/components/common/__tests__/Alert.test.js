import React from "react";
import { shallow } from "enzyme";

// Components
import Alert from "../Alert";

describe("Alert", () => {
  it("renders correctly", () => {
    let wrapper;
    wrapper = shallow(<Alert level={"warning"} text="warning" />);
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(<Alert level={"error"} text="error" />);
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(<Alert level={"info"} text="info" />);
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(<Alert level={"success"} text="success" />);
    expect(wrapper).toMatchSnapshot();
  });
});
