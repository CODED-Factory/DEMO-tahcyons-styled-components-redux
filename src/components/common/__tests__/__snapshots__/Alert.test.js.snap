// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  className="flex items-center justify-center pa3 br2 mv2 bg-light-yellow white"
>
  <span>
    warning
  </span>
</div>
`;

exports[`Alert renders correctly 2`] = `
<div
  className="flex items-center justify-center pa3 br2 mv2 bg-light-red white"
>
  <span>
    error
  </span>
</div>
`;

exports[`Alert renders correctly 3`] = `
<div
  className="flex items-center justify-center pa3 br2 mv2 bg-lightest-blue navy"
>
  <span>
    info
  </span>
</div>
`;

exports[`Alert renders correctly 4`] = `
<div
  className="flex items-center justify-center pa3 br2 mv2 bg-light-green dark-green"
>
  <span>
    success
  </span>
</div>
`;
